Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp

```diff
@@ -2081,4 +2135,85 @@ namespace EMStudio
+        if (eventPresetsTable == nullptr)
+        {
+            return false;
+        }
+
+        // get the number of motion event presets and iterate through them
+        const uint32 numRows = eventPresetsTable->rowCount();
+        for (uint32 i = 0; i < numRows; ++i)
+        {
+            QTableWidgetItem* itemType = eventPresetsTable->item(i, 1);
+
+            if (itemType->isSelected())
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+    void TimeViewPlugin::OnEventPresetDroppedOnTrackData(QPoint mousePoint)
+    {
+        EMotionFX::Motion* motion = GetCommandManager()->GetCurrentSelection().GetSingleMotion();
+        if (!motion)
+        {
+            return;
+        }
+
+        // calculate the start time for the motion event
+        double dropTimeInSeconds = PixelToTime(mousePoint.x());
+
+        // get the time track on which we dropped the preset
+        TimeTrack* timeTrack = GetTrackAt(mousePoint.y());
+        if (!timeTrack)
+        {
+            return;
+        }
+
+        // get the corresponding motion event track
+        EMotionFX::MotionEventTable* eventTable = motion->GetEventTable();
+        EMotionFX::MotionEventTrack* eventTrack = eventTable->FindTrackByName(timeTrack->GetName());
+        if (eventTrack == nullptr)
+        {
+            return;
+        }
+
+        // get the motion event presets table
+        QTableWidget* eventPresetsTable = m_motionEventsPresetsWidget->GetMotionEventPresetsTable();
+        if (eventPresetsTable == nullptr)
+        {
+            return;
+        }
+
+        // get the number of motion event presets and iterate through them
+        const size_t numRows = EMStudio::GetEventPresetManager()->GetNumPresets();
+        AZStd::string result;
+        for (size_t i = 0; i < numRows; ++i)
+        {
+            const MotionEventPreset* preset = EMStudio::GetEventPresetManager()->GetPreset(i);
+            QTableWidgetItem* itemName = eventPresetsTable->item(static_cast<int>(i), 1);
+
+            if (itemName->isSelected())
+            {
+                CommandSystem::CommandCreateMotionEvent* createMotionEventCommand = aznew CommandSystem::CommandCreateMotionEvent();
+                createMotionEventCommand->SetMotionID(motion->GetID());
+                createMotionEventCommand->SetEventTrackName(eventTrack->GetName());
+                createMotionEventCommand->SetStartTime(aznumeric_cast<float>(dropTimeInSeconds));
+                createMotionEventCommand->SetEndTime(aznumeric_cast<float>(dropTimeInSeconds));
+                createMotionEventCommand->SetEventDatas(preset->GetEventDatas());
+
+                if (!EMStudio::GetCommandManager()->ExecuteCommand(createMotionEventCommand, result))
+                {
+                    AZ_Error("EMotionFX", false, result.c_str());
+                }
+            }
```

Comment:
suggestion: This looks very similar to code in `MotionEventPresetsWidget::OnEventPresetDropped`, any chance we can unify this (move to a function?)

Reply:
Good catch!
The one in TimeViewPlugin can stay and do the work. I deleted the one in MotionEventPresets widget and connected the events to it.